libs/core/src/main/java/org/elasticsearch/common/util/concurrent/RefCounted.java

```diff
@@ -65,4 +65,15 @@
      * @return returns {@code true} if the ref count dropped to 0 as a result of calling this method
      */
     boolean decRef();
+
+    /**
+     * Decrement the ref count on the given {@code object} by one if it is a {@link RefCounted}
+     *
+     * @param object object to decrement ref count for if it is a {@link RefCounted}
+     */
+    static void decRef(Object object) {
```

Comment:
Would it be so bad to have `TransportMessage implements RefCounted` - I think it'll catch someone out in future. At least let's add Javadoc somewhere saying that subclasses that implement `RefCounted` are released by the transport service.

Reply:
I went with `TransportMessage implements RefCounted` after all. I initially didn't like the idea of doing that because it felt dirty to force overriding either none or all 3 methods from `RefCounted` when creating a ref-counted message ... but in hindsight that's still a lot better than implicitly releasing via the interface :)